nilearn/interfaces/bids/glm.py

```diff
@@ -0,0 +1,232 @@
+            contrast_plot = plot_contrast_matrix(
+                contrast_data,
+                design_matrix,
+                colorbar=True,
+            )
+            contrast_plot.set_xlabel(contrast_name)
+            contrast_plot.figure.set_figheight(2)
+            contrast_plot.figure.set_tight_layout(True)
+            contrast_name = _clean_contrast_name(contrast_name)
+            constrast_fig_file = os.path.join(
+                out_dir,
+                f'{prefix}{run_str}contrast-{contrast_name}_design.svg',
+            )
+            contrast_plot.figure.savefig(constrast_fig_file)
+
+    statistical_maps = _make_stat_maps(model, contrasts, output_type='all')
+
+    # Model metadata
+    # TODO: Determine optimal mapping of model metadata to BIDS fields.
+    metadata_file = os.path.join(out_dir, f'{prefix}statmap.json')
+    _generate_model_metadata(metadata_file, model)
+
+    dset_desc_file = os.path.join(out_dir, 'dataset_description.json')
+    _generate_dataset_description(dset_desc_file, model_level)
+
+    # Write out contrast-level statistical maps
+    for contrast_name, contrast_maps in statistical_maps.items():
+        # Extract stat_type
+        contrast_matrix = contrasts[contrast_name]
+        # Strings and 1D arrays are assumed to be t-contrasts
+        if isinstance(contrast_matrix, str) or (contrast_matrix.ndim == 1):
+            stat_type = 't'
+        else:
+            stat_type = 'F'
+
+        # Override automatic detection with explicit type if provided
+        stat_type = contrast_types.get(contrast_name, stat_type)
+
+        # Convert the contrast name to camelCase
+        contrast_name = _clean_contrast_name(contrast_name)
+
+        # Contrast-level images
+        contrast_level_mapping = {
+            'effect_size': (
+                f'{prefix}contrast-{contrast_name}_stat-effect_statmap.nii.gz'
+            ),
+            'stat': (
+                f'{prefix}contrast-{contrast_name}_stat-{stat_type}_statmap'
+                '.nii.gz'
+            ),
+            'effect_variance': (
+                f'{prefix}contrast-{contrast_name}_stat-variance_statmap'
+                '.nii.gz'
+            ),
+            'z_score': (
+                f'{prefix}contrast-{contrast_name}_stat-z_statmap.nii.gz'
+            ),
+            'p_value': (
+                f'{prefix}contrast-{contrast_name}_stat-p_statmap.nii.gz'
+            ),
+        }
+        # Rename keys
+        renamed_contrast_maps = {
+            contrast_level_mapping.get(k, k): v
+            for k, v in contrast_maps.items()
+        }
+
+        for map_name, img in renamed_contrast_maps.items():
+            out_file = os.path.join(out_dir, map_name)
+            img.to_filename(out_file)
+
+    # Write out model-level statistical maps
+    model_level_mapping = {
+        'residuals': f'{prefix}stat-errorts_statmap.nii.gz',
+        'r_square': f'{prefix}stat-rSquare_statmap.nii.gz',
```

Comment:
The names are primarily based on [the fitlins naming convention](https://github.com/poldracklab/fitlins/blob/c65b7d268c0f1c46c9ef6b9bf623590f107e077a/fitlins/workflows/base.py#L171). Some appear to be camelCase (rSquare, logLikelihood), while most are not.

I don't think the BIDS folks would have a problem with us changing these names, since they're not really codified. I would just want to also propose the change in my [pre-BEP document](https://docs.google.com/document/d/1OxKhhEctlQg6nmw1L3vw9Mm0K9-KkTpVKZuNqYnlYEM/edit?usp=sharing) to make sure that it's openly up for discussion.